### PR TITLE
Add menu that can create the game

### PR DESCRIPTION
**Pull request by KashaevaD**
Now you can create the single and multi player game.
If you create the single, you will start the game, look at 1 picture.
singlemenu

if you create the multi, you need to wait a second user, look at 2 picture .
createmulti

### CODE REVIEW

File: src/app/main-menu/mainmenu.service.ts [hunk 1]
```diff
@@ -1,22 +1,132 @@
...
+  private _getWordFromLanguage(item:number, wordId:number):string {
+    return (item % 2 === 0)?  this._firstLanguageArray[wordId] :  this._lastLanguageArray[wordId];
+  }
+
+  private _generateIdRoom():number {
+    return new Date().getTime();
   }
 
 
   // work with localStorage
```
Comment (stevermeister) [inline src/app/main-menu/mainmenu.service.ts]: it should not be here

File: src/app/main-menu/mainmenu.service.ts [hunk 2]
```diff
@@ -1,22 +1,132 @@
 import { Injectable, EventEmitter } from '@angular/core';
 import { SIZE } from '../constants';
+import { Subscription } from "rxjs";
+import {AngularFire, FirebaseObjectObservable} from 'angularfire2';
+
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/switchMap';
 
 @Injectable()
 export class MainmenuService {
```
Comment (stevermeister) [inline src/app/main-menu/mainmenu.service.ts]: main menu? how is it related to menu?

File: src/app/main-menu/mainmenu.service.ts (hunk 1 above)
Comment (KashaevaD) [inline src/app/main-menu/mainmenu.service.ts]: Why do you think so? It is service in our mainmenu component, or I need to move it to single service?

File: src/app/main-menu/main-menu.component.ts [hunk 3]
```diff
@@ -26,6 +32,26 @@ export class MainMenuComponent {
       this.isMainMenuPage = result;
     });
 
+    //if user created a multiplayer game
+    this._waitForUserSubscriber = this._menuService.waitForSecondUserMultiplayer.subscribe((id) => {
+      this.isWait = true;
+
+      let base = document.querySelector('base').getAttribute("href") || "/";
+      this.shareAbleLink = window.location.origin.concat(base, "mainmenu/multi/", id);
```
Comment (stevermeister) [inline src/app/main-menu/main-menu.component.ts]: ugly
should be replaced in separate service

File: src/app/main-menu/mainmenu.service.ts (hunk 2 above)
Comment (KashaevaD) [inline src/app/main-menu/mainmenu.service.ts]: it is  a main service our menu, at this service we can create room on firebase and form our card

File: src/app/main-menu/mainmenu.service.ts [hunk 4]
```diff
@@ -1,22 +1,132 @@
...
 
+  public startPlayingGame:EventEmitter<number>;
+  public waitForSecondUserMultiplayer: EventEmitter<number>;
 
-  constructor() {
+
+  constructor(private _af : AngularFire) {
     this.isHideIntroForUser = new EventEmitter();
+    this.startPlayingGame = new EventEmitter();
+    this.waitForSecondUserMultiplayer = new EventEmitter();
```
Comment (stevermeister) [inline src/app/main-menu/mainmenu.service.ts]: why EventEmitter but not Subject ?

**Pull request merged**